extensions/sql-database-projects/src/models/project.ts

```diff
@@ -347,13 +347,10 @@ export class Project {
 	public async deleteFileFolder(entry: FileProjectEntry): Promise<void> {
 		// compile a list of folder contents to delete; if entry is a file, contents will contain only itself
 		const toDeleteFiles: FileProjectEntry[] = this.files.concat(this.preDeployScripts).concat(this.postDeployScripts).concat(this.noneDeployScripts).filter(x => x.fsUri.fsPath.startsWith(entry.fsUri.fsPath) && x.type === EntryType.File);
-		const toDeleteFolders: FileProjectEntry[] = this.files.filter(x => x.fsUri.fsPath.startsWith(entry.fsUri.fsPath) && x.type === EntryType.Folder).sort(x => -x.relativePath.length);
+		const toDeleteFolders: FileProjectEntry[] = this.files.filter(x => x.fsUri.fsPath.startsWith(entry.fsUri.fsPath) && x.type === EntryType.Folder);
```

Comment:
Did you try the scenario where there are multiple folder levels to be deleted?
A hierarchy like this:
![image](https://user-images.githubusercontent.com/57200045/111521763-be7faa00-8716-11eb-8fcd-9ecda47aaa2d.png)

If MessageQueueActivate.sql is excluded, then Storedprocs as well dbo folder should be excluded.

Reply:
Yes, that's actually why I made the test case so complicated. I'll try to capture this scenario as I simplify the test.